scenario_player/utils/legacy.py

```diff
@@ -222,6 +222,9 @@ def wait_for_txs(
                 status = tx.get("status")
                 if status is not None and status == 0:
                     raise ScenarioTxError(f"Transaction {txhash} failed.")
+                ## we want to add 2 blocks as confirmation
+                if tx["blockNumber"] + 2 >= web3.eth.getBlock("latest")['number']:
+                    continue
                 txhashes.remove(txhash)
```

Comment:
I would invert the condition and move the `txhashes.remove()` call into the condition. 
As it's currently the `continue` will also skip the `sleep` and thereby spam the eth node with requests.

Reply:
true